changelog/3107.feature

```diff
@@ -0,0 +1 @@
+Optimize reorder_items in fixtures.py.
```

Comment:
This isn't really an useful changelog for a *user* of pytest to read, IMHO. What about something like:

> Improve performance when collecting tests using many fixtures.

or so?

Reply:
Sounds good to me.

Reply:
@The-Compiler I added the requested change, is there anything else I need to do?  I'm not an experienced githubber.

Reply:
No, other than wait for someone to review this. I unfortunately don't have the time right now, but probably others will take a look soon :smile: